### PR TITLE
Drop the use of the deprecated setuptools Features

### LINKED ISSUE #11: setup.py: Features are deprecated
When installing version 4.2.0 I get the following warnings:

```
Getting distribution for 'zope.hookable==4.2.0'.
/tmp/easy_install-7rWmxt/zope.hookable-4.2.0/setup.py:70: DistDeprecationWarning: Features are deprecated and will be removed in a future version. See https://github.com/pypa/setuptools/issues/65.
  [os.path.join('src', 'zope', 'hookable', "_zope_hookable.c")],
/usr/local/lib/python2.7/distutils/core.py:111: DistDeprecationWarning: Features are deprecated and will be removed in a future version. See https://github.com/pypa/setuptools/issues/65.
  _setup_distribution = dist = klass(attrs)
```

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #11

https://github.com/pypa/setuptools/issues/65

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -64,26 +63,22 @@ def read(*rnames):
...
-    ],
-)
+]
 
 is_pypy_or_jython = platform.python_implementation() in ('PyPy', 'Jython')
 
 # Jython cannot build the C optimizations, while on PyPy they are
 # anti-optimizations (the C extension compatibility layer is known-slow,
 # and defeats JIT opportunities).
 if is_pypy_or_jython:
```
Comment (mgedmin) [inline setup.py]: Maybe please add an or is_pure check with the os.getenv('PURE_PYTHON') logic copied from zope.interface etc.? So people can ask to skip extension modules.

Comment (jamadden) [reply setup.py]: @mgedmin Based on what we decided in zope.interface and zope.i18nmessageid, is there anything that still needs done here?

Comment (mgedmin) [reply setup.py]: A changelog entry would be nice!

**Comment by mgedmin** [comment]
(I'm going to go AFK for a couple of days in about 5 minutes, so feel free to make a release after you merge this -- we won't conflict.)

**Pull request merged**